sonarts-sq-plugin/its/plugin/src/test/java/org/sonar/typescript/its/JavaRulesIntegrationTest.java

```diff
@@ -0,0 +1,271 @@
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.zip.ZipEntry;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameter;
+import org.junit.runners.Parameterized.Parameters;
+import org.sonar.api.batch.fs.InputFile;
+import org.sonar.api.batch.fs.internal.TestInputFileBuilder;
+import org.sonar.api.batch.rule.ActiveRules;
+import org.sonar.api.batch.rule.CheckFactory;
+import org.sonar.api.batch.rule.Checks;
+import org.sonar.api.batch.rule.internal.ActiveRulesBuilder;
+import org.sonar.api.batch.rule.internal.NewActiveRule;
+import org.sonar.api.batch.sensor.internal.SensorContextTester;
+import org.sonar.api.batch.sensor.issue.Issue;
+import org.sonar.api.config.Configuration;
+import org.sonar.api.issue.NoSonarFilter;
+import org.sonar.api.measures.FileLinesContext;
+import org.sonar.api.measures.FileLinesContextFactory;
+import org.sonar.api.rule.RuleKey;
+import org.sonar.check.Rule;
+import org.sonar.plugin.typescript.ExternalProcessStreamConsumer;
+import org.sonar.plugin.typescript.ExternalTypescriptSensor;
+import org.sonar.plugin.typescript.TypeScriptLanguage;
+import org.sonar.plugin.typescript.TypeScriptRules;
+import org.sonar.plugin.typescript.TypeScriptRulesDefinition;
+import org.sonar.plugin.typescript.executable.ExecutableBundle;
+import org.sonar.plugin.typescript.executable.ExecutableBundleFactory;
+import org.sonar.plugin.typescript.executable.SonarTSCoreBundle;
+import org.sonar.plugin.typescript.executable.Zip;
+import org.sonar.plugin.typescript.rules.TypeScriptRule;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+
+/**
+ * This test reads all the available rules in the SQ plugin via reflection and runs the plugin on the test file used for sonarts-core unit test (*.lint.ts file)
+ */
+@RunWith(Parameterized.class)
+public class JavaRulesIntegrationTest {
+
+  private static final Path SONARTS_LINT_FILES = Paths.get("../../../sonarts-core/tests/rules");
+  private static final Path TEST_FIXTURES = Paths.get("src/test/resources/fixtures");
+
+  private static final Pattern ERROR_COMMENT = Pattern.compile(".*//.*\\^+\\s*\\{\\{.*}}.*");
+  private static final Pattern MULTILINE_ERROR_COMMENT = Pattern.compile(".*//\\s*\\[(\\d+):(\\d+)-(\\d+):(\\d+)]\\s*\\{\\{.*");
+  private static final String FIXTURE_FILENAME = "fixture.ts";
+  private static final byte[] TSCONFIG = ("{" +
+    "  \"compilerOptions\": {\n" +
+    "    \"strict\": true,\n" +
+    "    \"target\": \"esnext\"\n" +
+    "  },\n" +
+    "  \"include\": [\"" + FIXTURE_FILENAME + "\"]\n" +
+    "}").getBytes(StandardCharsets.UTF_8);
+
+  // for some rules we don't have test file available, skip those
+  private static final Set<String> MISSING_TEST_FILE = new HashSet<>(Arrays.asList(
```

Comment:
@saberduck I had 46 tests executed, but there are 48 rules in tslint-sonarts, plus 1 tested rule from tslint inself: there are 3 rules lost somewhere